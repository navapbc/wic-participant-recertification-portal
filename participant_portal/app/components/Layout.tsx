import { Trans } from "react-i18next";
import type { ReactElement } from "react";
import { Alert, Title, Header, Footer } from "@trussworks/react-uswds";
import TransLinks from "./TransLinks";

export type LayoutProps = {
  children: ReactElement;
  demoMode?: string;
  missingData?: string;
};

const Layout = ({
  children,
  demoMode,
  missingData,
}: LayoutProps): ReactElement => {
  return (
    <div className="container">
      {demoMode === "true" ? (
        <Alert type="warning" headingLevel="h6" slim={true} role="status">
          <Trans i18nKey={"Demo.alertText"} />
        </Alert>
      ) : (
        ""
      )}
      <Header extended>
        <div className="usa-navbar">
          <Title id="extended-logo">
            <em className="usa-logo__text text-primary-darker">
              <Trans i18nKey="Layout.header" />
            </em>
          </Title>
        </div>
      </Header>
      <main className="main">
        <div className="grid-row">
          <div className="desktop:grid-col-8 padding-2 padding-bottom-8">
            {missingData === "true" ? (
              <Alert
                type="error"
                headingLevel="h4"
                className="margin-bottom-3"
                role="alert"
              >
                <Trans i18nKey={"routingError"} />
              </Alert>
            ) : (
              ""
            )}
          </div>
        </div>
        {children}
      </main>
      <Footer
        size="slim"
        primary=""
        secondary={
          <TransLinks
            i18nTextKey="Layout.footer.text"
            i18nLinkKey="Layout.footer.links"
          />
        }
      />
    </div>
  );
};

export default Layout;
