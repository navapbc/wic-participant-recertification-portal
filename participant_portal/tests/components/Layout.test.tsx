import Layout from "app/components/Layout";
import { renderWithRouter } from "tests/helpers/setup";

import { testAccessibility } from "../helpers/sharedTests";

it("should match snapshot", () => {
  const { container } = renderWithRouter(
    <Layout children={<h1>'child'</h1>} />
  );
  expect(container).toMatchSnapshot();
});

// eslint-disable-next-line jest/expect-expect
it("should pass accessibility scan", async () => {
  // testAccessibility does have an assertion (expect(results).toHaveNoViolations)
  await testAccessibility(<Layout children={<h1>'child'</h1>} />);
});
