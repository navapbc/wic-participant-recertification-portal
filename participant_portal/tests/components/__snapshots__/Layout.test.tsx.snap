// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <header
      class="usa-header usa-header--extended"
      data-testid="header"
    >
      <div
        class="usa-navbar"
      >
        <div
          class="usa-logo"
          id="extended-logo"
        >
          <em
            class="usa-logo__text"
          >
            <em
              class="usa-logo__text text-primary-darker"
            >
              Shoegaze Stack
            </em>
          </em>
        </div>
      </div>
    </header>
    <main
      class="main"
    >
      <div
        class="grid-row"
      >
        <div
          class="desktop:grid-col-8 padding-2 padding-bottom-8"
        />
      </div>
      <h1>
        'child'
      </h1>
    </main>
    <footer
      class="usa-footer usa-footer--slim"
    >
      <div
        class="usa-footer__primary-section"
      />
      <div
        class="usa-footer__secondary-section"
      >
        <div
          class="grid-container"
        >
          For more information, read the 
          <a
            class="usa-link usa-link--external"
            href="https://remix.run"
            rel="noopener noreferrer"
            target="_blank"
          >
            Remix Run Documentation
          </a>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
